### PR TITLE
[WIP] Correct regularization in DICS

### PR DESCRIPTION
**Pull request by brovelli**
Note that I created a Pull Request just for _dics.py
If such regularization is correct, we should do the same modifications in _lcmv.py

### CODE REVIEW

File: mne/beamformer/_dics.py [hunk 1]
```diff
@@ -66,9 +66,12 @@ def _apply_dics(data, info, tmin, forward, noise_csd, data_csd, reg,
 
     Cm = data_csd.data
 
-    # Calculating regularized inverse, equivalent to an inverse operation after
-    # regularization: Cm += reg * np.trace(Cm) / len(Cm) * np.eye(len(Cm))
-    Cm_inv = linalg.pinv(Cm, reg)
+    # Tikhonov regularization using reg parameter to control for trade-off between spatial resolution and noise sensitivity
```
Comment (mmagnuski) [inline mne/beamformer/_dics.py]: this line is too long. mne has pep8 tests and they allow only lines <= 79 characters

File: mne/beamformer/_dics.py [hunk 2]
```diff
@@ -379,10 +382,12 @@ def csd_shapes(x):
 
         Cm = data_csd.data
 
-        # Calculating regularized inverse, equivalent to an inverse operation
-        # after the following regularization:
-        # Cm += reg * np.trace(Cm) / len(Cm) * np.eye(len(Cm))
-        Cm_inv = linalg.pinv(Cm, reg)
+        # Tikhonov regularization using reg parameter to control for trade-off between spatial resolution and noise sensitivity
```
Comment (mmagnuski) [inline mne/beamformer/_dics.py]: same here

**Comment by brovelli** [comment]
Now it should be fine

**Comment by agramfort** [comment]
thanks @brovelli for taking a stab as this.

let me know if you need help and set the PR to MRG when CIs are happy.

**Comment by brovelli** [comment]
In fact, I don't have a clue of how to set the PR to MRG. Can someone 
else do that please?

thanks

**Comment by mmagnuski** [comment]
@brovelli you just change the title of the PR from starting with `[WIP]` to `[MRG]`

**Comment by mmagnuski** [comment]
BTW - two DICS tests fail:

```
======================================================================
FAIL: Test DICS with evoked data and single trials
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/build/mne-tools/mne-python/mne/beamformer/tests/test_dics.py", line 106, in test_dics
    assert_true(0.04 < tmax < 0.05)
AssertionError: False is not true
    'False is not true' = self._formatMessage('False is not true', "%s is not true" % safe_repr(False))
>>  raise self.failureException('False is not true')
    
======================================================================
FAIL: Test DICS source power computation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/test/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/build/mne-tools/mne-python/mne/beamformer/tests/test_dics.py", line 175, in test_dics_source_power
    assert_true(max_source_idx == 0)
AssertionError: False is not true
    'False is not true' = self._formatMessage('False is not true', "%s is not true" % safe_repr(False))
>>  raise self.failureException('False is not true')
```

**Comment by brovelli** [comment]
Sorry I cannot help on this

**Comment by agramfort** [comment]
see https://github.com/mne-tools/mne-python/pull/3854

closing

thx @brovelli

**Comment by brovelli** [comment]
Thanks to you guys

**Pull request closed without merging**